app/middleware/session_manager.py

```diff
@@ -33,28 +33,28 @@ def __init__(self, db_session):
         self._session_factory = db_session
         self._scoped = isinstance(db_session, scoping.ScopedSession)
 
-    def process_request(self, req, res, resource=None):
+    def process_request(self, req, res):
         """
         Handle post-processing of the response (after routing).
         """
         req.context['session'] = self._session_factory
 
-    def process_response(self, req, res, resource=None):
+    def process_response(self, req, res, resource=None, req_succeeded=None):
         """
         Handle post-processing of the response (after routing).
         """
-        session = req.context['session']
-
-        if config.DB_AUTOCOMMIT:
-            try:
-                session.commit()
-            except SQLAlchemyError as ex:
-                session.rollback()
-                raise DatabaseError(ERR_DATABASE_ROLLBACK, ex.args, ex.params)
-
-        if self._scoped:
-            # remove any database-loaded state from all current objects
-            # so that the next access of any attribute, or any query execution will retrieve new state
-            session.remove()
-        else:
-            session.close()
+        session = req.context.get('session')
```

Comment:
If an error occurs in another Middleware process_request, it will not be executed this process_request.
In that case, 'session' is not set in context and an error occurs here.